Load the data and divide into training and test.

In [1]:
import pandas as pd

df = pd.read_csv("https://data.heatonresearch.com/data/t81-558/datasets/series-31-num.csv")

In [15]:
df_train = df[df['time']<3000]
df_test = df[df['time']>=3000]

seq_train = df_train['value'].tolist()
seq_test = df_test['value'].tolist()

See size of each:

In [17]:
print(f"Size of test: {len(seq_test)}")

Size of test: 1000


In [9]:
import numpy as np
def to_sequences(seq_size, obs):
    x = []
    y = []
    for i in range(len(obs)-SEQUENCE_SIZE):
        window = obs[i:(i+SEQUENCE_SIZE)]
        after_window = obs[i+SEQUENCE_SIZE]
        window = [[x] for x in window]
        x.append(window)
        y.append(after_window)
        
    return np.array(x),np.array(y)

Now divide into sequences.

In [10]:
SEQUENCE_SIZE = 5
x_train,y_train = to_sequences(SEQUENCE_SIZE,seq_train)
x_test,y_test = to_sequences(SEQUENCE_SIZE,seq_test)

In [18]:
x_test.shape

(995, 5, 1)

The test set that I produced, using the provided to_sequences function returns 995 sequences.  This is because to_sequences expects 5 inputs (for x) plus one expected valye (for y).  This is actually wasting one value, because we do not need the y-value for evaluation.

If you use your own method to generate the test data, you could get 996 rows, because your final x row will include the final value in the dataset (15). 

For this one I will accept either 995 or 996.